any_test_eval_model.py

```diff
@@ -26,131 +26,60 @@
-    output_dir = args.output_dir
-    dtype=dtype_from_args(args.datatype)
-
-    # --------- Load data
     dataset_names = ["test"]
-    if args.use_dc2:
-        datadir = "./tests/deepdisc/test_data/dc2/"
-    else:
-        datadir = "/home/shared/hsc/HSC/HSC_DR3/data/"
     t0 = time.time()
     dataset_dicts = {}
     for i, d in enumerate(dataset_names):
         dataset_dicts[d] = get_data_from_json(testfile)
     print("Took ", time.time() - t0, "seconds to load samples")
     
-    # Local vars/metadata
-    #classes = ["star", "galaxy"]
-    bb = args.run_name.split("_")[0] # backbone
-    
-    # --------- Start config stuff
-    cfgfile = (
-        f"./tests/deepdisc/test_data/configs/"
-        f"solo/solo_cascade_mask_rcnn_swin_b_in21k_50ep_test_eval.py"
-    )
-    cfg = LazyConfig.load(cfgfile)
-    
-    # --------- Setting a bunch of config stuff
-    cfg.OUTPUT_DIR = output_dir
-    
-    cfg.model.roi_heads.num_classes = args.nc
-
-    for bp in cfg.model.roi_heads.box_predictors:
-        bp.test_score_thresh = roi_thresh
+    return dataset_dicts
 
-    for box_predictor in cfg.model.roi_heads.box_predictors:
-        box_predictor.test_topk_per_image = 1000
-        box_predictor.test_score_thresh = roi_thresh
 
-    cfg.train.init_checkpoint = os.path.join(output_dir, run_name)
+def get_config(use_dc2, use_redshift, output_dir, run_name):
+    """Get the relevant config based on if using dc2/redshifts.
     
-    if args.use_dc2:
-        cfg.model.backbone.bottom_up.in_chans = 6
-        cfg.model.pixel_mean = [0.05381286, 0.04986344, 0.07526361, 0.10420945, 0.14229655, 0.21245764]
-        cfg.model.pixel_std = [2.9318833, 1.8443471, 2.581817, 3.5950038, 4.5809164, 7.302009]
-        
-        if args.use_redshift:
-            cfg.model.roi_heads.num_components=5
-            cfg.model.roi_heads._target_ = RedshiftPDFCasROIHeads
-            #cfg.zloss_factor = 1.0
-            #cfg.model.zloss_factor = 1.0
-            cfg.model.roi_heads.zloss_factor = 1.0 #! what's a reasonable default?
+    Adds the MISC keys into the the top level of the config
+    (these would otherwise be ignored as config file local vars when importing)
+    """
+    cfg_dir = "./tests/deepdisc/test_data/configs/solo"
+    if use_dc2:
+        if use_redshift:
+            file_name = "solo_cascade_mask_rcnn_swin_b_in21k_50ep_test_eval_DC2_redshift.py"
+        file_name = "solo_cascade_mask_rcnn_swin_b_in21k_50ep_test_eval_DC2.py"
+    else:
+        file_name = "solo_cascade_mask_rcnn_swin_b_in21k_50ep_test_eval.py"
+    cfg_file = f"{cfg_dir}/{file_name}"
+    cfg = LazyConfig.load(cfg_file)
     
-        #! this maybe shouldn't have been a config value? or should we make a sep config for dc2?
-        cfg.classes = ["object"] 
-        
-    # --------- Now we case predictor on model type, and if using dc2 data
+    # Set misc vals as top level vals
+    for key in cfg.get("MISC", dict()).keys():
+        cfg[key] = cfg.MISC[key]
```

Comment:
I think you could also do something like this here: `cfg.update(cfg.get("MISC", dict()))`. But what you have is more explicit, so probably best to keep it as is.